File: JMImoveis.API/Entities/LeadPostVisit.cs
namespace JMImoveisAPI.Entities
{
    public class LeadPostVisit
    {
        public long Id { get; set; }
        public int LeadId { get; set; }
        public string? Cpf { get; set; }
        public bool? HasRestriction { get; set; }
        public string? IncomeType { get; set; }
        public string? InterestRegion { get; set; }
        public bool? PaysRent { get; set; }
        public string? MaritalStatus { get; set; }
        public decimal? DownPaymentAmount { get; set; }
        public long? AttendingAgentId { get; set; }
        public string? PropertyInterestType { get; set; }
        public string PostVisitStatus { get; set; } = "ACOMPANHANDO";
        public DateTime? NextFollowUpAt { get; set; }
        public string? LastInteractionSummary { get; set; }
        public long? ProposalId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }

    public class LeadPostVisitRequest
    {
        public string? Cpf { get; set; }
        public bool? HasRestriction { get; set; }
        public string? IncomeType { get; set; }
        public string? InterestRegion { get; set; }
        public bool? PaysRent { get; set; }
        public string? MaritalStatus { get; set; }
        public decimal? DownPaymentAmount { get; set; }
        public long? AttendingAgentId { get; set; }
        public string? PropertyInterestType { get; set; }
        public string? PostVisitStatus { get; set; }
        public DateTime? NextFollowUpAt { get; set; }
        public string? LastInteractionSummary { get; set; }
        public long? ProposalId { get; set; }
    }

    public class UpdateLeadPostVisitStatusRequest
    {
        public string Status { get; set; } = string.Empty;
    }

    public class LeadPostVisitListItem
    {
        public long PostVisitId { get; set; }
        public int LeadId { get; set; }
        public string NomeCliente { get; set; } = string.Empty;
        public string? Telefone { get; set; }
        public string? Email { get; set; }
        public string? Cpf { get; set; }
        public string PostVisitStatus { get; set; } = string.Empty;
        public DateTime? NextFollowUpAt { get; set; }
        public long? AttendingAgentId { get; set; }
        public string? AttendingAgentName { get; set; }
        public string? InterestRegion { get; set; }
        public decimal? DownPaymentAmount { get; set; }
        public string? PropertyInterestType { get; set; }
        public string? LastInteractionSummary { get; set; }
        public long? ProposalId { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: JMImoveis.Web/src/app/models/lead-post-visit.ts
export type LeadPostVisitStatus =
  | 'AGENDOU_RETORNO'
  | 'OPORTUNIDADE_FUTURA'
  | 'ACOMPANHANDO'
  | 'EM_PROPOSTA'
  | 'FECHOU_VENDA';

export interface LeadPostVisitListItem {
  postVisitId: number;
  leadId: number;
  nomeCliente: string;
  telefone?: string | null;
  email?: string | null;
  cpf?: string | null;
  postVisitStatus: LeadPostVisitStatus;
  nextFollowUpAt?: string | null;
  attendingAgentId?: number | null;
  attendingAgentName?: string | null;
  interestRegion?: string | null;
  downPaymentAmount?: number | null;
  propertyInterestType?: string | null;
  lastInteractionSummary?: string | null;
  proposalId?: number | null;
  createdAt: string;
  updatedAt?: string | null;
}

export interface LeadPostVisit {
  id: number;
  leadId: number;
  cpf?: string | null;
  hasRestriction?: boolean | null;
  incomeType?: string | null;
  interestRegion?: string | null;
  paysRent?: boolean | null;
  maritalStatus?: string | null;
  downPaymentAmount?: number | null;
  attendingAgentId?: number | null;
  propertyInterestType?: string | null;
  postVisitStatus: LeadPostVisitStatus;
  nextFollowUpAt?: string | null;
  lastInteractionSummary?: string | null;
  proposalId?: number | null;
  createdAt: string;
  updatedAt?: string | null;
  deletedAt?: string | null;
}

export interface LeadPostVisitRequest {
  cpf?: string | null;
  hasRestriction?: boolean | null;
  incomeType?: string | null;
  interestRegion?: string | null;
  paysRent?: boolean | null;
  maritalStatus?: string | null;
  downPaymentAmount?: number | null;
  attendingAgentId?: number | null;
  propertyInterestType?: string | null;
  postVisitStatus?: LeadPostVisitStatus | null;
  nextFollowUpAt?: string | null;
  lastInteractionSummary?: string | null;
  proposalId?: number | null;
}

export interface LeadPostVisitFilters {
  status?: LeadPostVisitStatus | '' | null;
  agentId?: number | string | null;
  search?: string | null;
  followUpFrom?: string | null;
  followUpTo?: string | null;
}
